bevy_lint/docs/README.md

```diff
@@ -0,0 +1,64 @@
+        - ⭐️ [`snippet()`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/source/fn.snippet.html)
+    - [`rustc_driver`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_driver/index.html)
+        - [`RunCompiler`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_driver/struct.RunCompiler.html)
+        - [`Callbacks`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_driver/trait.Callbacks.html)
+    - [`rustc_hir`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/index.html)
+        - ⭐️ [`DefId`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def_id/struct.DefId.html)
+        - [`LocalDefId`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def_id/struct.LocalDefId.html)
+        - [`HirId`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir_id/struct.HirId.html)
+        - ⭐️ [`Expr`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.Expr.html)
+        - [`Item`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.Item.html)
+        - [`Path`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.Path.html)
+        - [`Ty`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/hir/struct.Ty.html) (Not to be confused with `rustc_middle::ty::Ty`)
+    - [`rustc_span`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/index.html)
+    - [`rustc_lint`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/index.html)
+        - ⭐️ [`LateLintPass`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/trait.LateLintPass.html)
+        - ⭐️ [`LateContext`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/struct.LateContext.html)
+    - [`rustc_lint_defs`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint_defs/index.html)
+    - [`rustc_middle`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/index.html)
+        - ⭐️ [`TyCtxt`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html)
+        - ⭐️ [`Ty`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Ty.html) (Not to be confused with HIR `Ty`)
+        - [`Map`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/hir/map/struct.Map.html) (HIR)
+- [Clippy Development](https://doc.rust-lang.org/stable/clippy/development/index.html)
+    - ⭐️ [Lint passes](https://doc.rust-lang.org/stable/clippy/development/lint_passes.html)
+    - ⭐️ [Emitting a lint](https://doc.rust-lang.org/stable/clippy/development/emitting_lints.html)
+    - ⭐️ [Type Checking](https://doc.rust-lang.org/stable/clippy/development/type_checking.html)
+    - ⭐️ [Dealing with macros and expansions](https://doc.rust-lang.org/stable/clippy/development/macro_expansions.html)
+    - [Common tools for writing lints](https://doc.rust-lang.org/stable/clippy/development/common_tools_writing_lints.html)
```

Comment:
```suggestion
    - [Common tools for writing lints](https://doc.rust-lang.org/stable/clippy/development/common_tools_writing_lints.html)
    - [Cargo lints](https://doc.rust-lang.org/stable/clippy/development/adding_lints.html#cargo-lints)
```

Reply:
Unfortunately I think this link may also be too Clippy-specific. What are your thoughts on waiting on this until we have actual Cargo lints (like #185, if it uses `cargo metadata`!), then writing `bevy_lint` specific versions of it. (I can write this, you just need to lay out the foundation for how Cargo lints would work here.)

Reply:
Yeah, it is very clippy specific I'm fine with waiting until we can reference ourselves!